File: imagepipeline/src/test/java/com/facebook/imagepipeline/core/ImagePipelineConfigTest.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package com.facebook.imagepipeline.core;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock;

import android.net.Uri;
import com.facebook.imagepipeline.request.ImageRequest;
import com.facebook.imagepipeline.request.ImageRequestBuilder;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.MockitoAnnotations;
import org.robolectric.RobolectricTestRunner;

/**
 * Some tests for ImagePipelineConfigTest
 */
@RunWith(RobolectricTestRunner.class)
public class ImagePipelineConfigTest {

  private Uri mUri;

  @Before
  public void setUp() {
    MockitoAnnotations.initMocks(this);
    mUri = mock(Uri.class);
  }

  @Test
  public void testDefaultConfigIsFalseByDefault() {
    ImagePipelineConfig.resetDefaultRequestConfig();
    assertFalse(ImagePipelineConfig.getDefaultImageRequestConfig().isProgressiveRenderingEnabled());
  }

  @Test
  public void testDefaultConfigIsTrueIfChanged() {
    ImagePipelineConfig.resetDefaultRequestConfig();
    ImagePipelineConfig.getDefaultImageRequestConfig().setProgressiveRenderingEnabled(true);
    assertTrue(ImagePipelineConfig.getDefaultImageRequestConfig().isProgressiveRenderingEnabled());
  }

  @Test
  public void testImageRequestDefault() {
    ImagePipelineConfig.resetDefaultRequestConfig();
    final ImageRequest imageRequest = ImageRequestBuilder.newBuilderWithSource(mUri).build();
    assertFalse(imageRequest.getProgressiveRenderingEnabled());
  }

  @Test
  public void testImageRequestWhenChanged() {
    ImagePipelineConfig.resetDefaultRequestConfig();
    ImagePipelineConfig.getDefaultImageRequestConfig().setProgressiveRenderingEnabled(true);
    final ImageRequest imageRequest = ImageRequestBuilder.newBuilderWithSource(mUri).build();
    assertTrue(imageRequest.getProgressiveRenderingEnabled());
  }

  @Test
  public void testImageRequestWhenChangedAndOverriden() {
    ImagePipelineConfig.resetDefaultRequestConfig();
    final ImageRequest imageRequest = ImageRequestBuilder.newBuilderWithSource(mUri)
        .setProgressiveRenderingEnabled(true)
        .build();
    assertTrue(imageRequest.getProgressiveRenderingEnabled());
    final ImageRequest imageRequest2 = ImageRequestBuilder.newBuilderWithSource(mUri)
        .setProgressiveRenderingEnabled(false)
        .build();
    assertFalse(imageRequest2.getProgressiveRenderingEnabled());
  }
}

File: imagepipeline/src/test/java/com/facebook/imagepipeline/core/ImagePipelineConfigTest.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.imagepipeline.core

import android.net.Uri
import com.facebook.imagepipeline.core.ImagePipelineConfig.Companion.defaultImageRequestConfig
import com.facebook.imagepipeline.core.ImagePipelineConfig.Companion.resetDefaultRequestConfig
import com.facebook.imagepipeline.request.ImageRequestBuilder
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito
import org.mockito.MockitoAnnotations
import org.robolectric.RobolectricTestRunner

/** Some tests for ImagePipelineConfigTest */
@RunWith(RobolectricTestRunner::class)
class ImagePipelineConfigTest {
  private lateinit var uri: Uri

  @Before
  fun setUp() {
    MockitoAnnotations.initMocks(this)
    uri = Mockito.mock(Uri::class.java)
  }

  @Test
  fun testDefaultConfigIsFalseByDefault() {
    resetDefaultRequestConfig()
    assertThat(defaultImageRequestConfig.isProgressiveRenderingEnabled).isFalse()
  }

  @Test
  fun testDefaultConfigIsTrueIfChanged() {
    resetDefaultRequestConfig()
    defaultImageRequestConfig.isProgressiveRenderingEnabled = true
    assertThat(defaultImageRequestConfig.isProgressiveRenderingEnabled).isTrue()
  }

  @Test
  fun testImageRequestDefault() {
    resetDefaultRequestConfig()
    val imageRequest = ImageRequestBuilder.newBuilderWithSource(uri).build()
    assertThat(imageRequest.getProgressiveRenderingEnabled()).isFalse()
  }

  @Test
  fun testImageRequestWhenChanged() {
    resetDefaultRequestConfig()
    defaultImageRequestConfig.isProgressiveRenderingEnabled = true
    val imageRequest = ImageRequestBuilder.newBuilderWithSource(uri).build()
    assertThat(imageRequest.getProgressiveRenderingEnabled()).isTrue()
  }

  @Test
  fun testImageRequestWhenChangedAndOverriden() {
    resetDefaultRequestConfig()
    val imageRequest =
        ImageRequestBuilder.newBuilderWithSource(uri).setProgressiveRenderingEnabled(true).build()
    assertThat(imageRequest.getProgressiveRenderingEnabled()).isTrue()
    val imageRequest2 =
        ImageRequestBuilder.newBuilderWithSource(uri).setProgressiveRenderingEnabled(false).build()
    assertThat(imageRequest2.getProgressiveRenderingEnabled()).isFalse()
  }
}
